src/contexts/ToolOptionsContext.tsx

```diff
@@ -0,0 +1,33 @@
+import {MediaToolOptions} from '@types'
+import React, {PropsWithChildren, createContext, useContext} from 'react'
+import {DropzoneOptions} from 'react-dropzone'
+
+type ContextProps = {
+  dropzone: Pick<DropzoneOptions, 'maxSize'>
+}
+
+const ToolOptionsContext = createContext<ContextProps | null>(null)
+
+type Props = {
+  options?: MediaToolOptions
+}
+
+export const ToolOptionsProvider = ({options, children}: PropsWithChildren<Props>) => {
+  return (
+    <ToolOptionsContext.Provider value={{dropzone: {maxSize: options?.maximumUploadSize}}}>
```

Comment:
We should memoize the context value to avoid rerendering context dependants needlessly.

Reply:
Ah, great catch! Fixed.